src/gpio.rs

```diff
@@ -42,6 +41,26 @@ pub struct Output<MODE> {
     _mode: PhantomData<MODE>,
 }
 
+/// Alternate function
+pub struct Alternate<MODE> {
+    _mode: PhantomData<MODE>,
+}
+
+/// Alternate Function 1
+pub struct Func1;
```

Comment:
Slightly superficial, but I don't really like `Func1`, could we have `AF1` etc instead?